ducktape/command_line/main.py

```diff
@@ -102,12 +43,56 @@ def extend_import_paths(paths):
     """
     for path in paths:
         dir = os.path.abspath(path if os.path.isdir(path) else os.path.dirname(path))
-        while(os.path.exists(os.path.join(dir, '__init__.py'))):
+        while os.path.exists(os.path.join(dir, '__init__.py')):
             dir = os.path.dirname(dir)
         sys.path.append(dir)
 
 
-def setup_results_directory(results_root, new_results_dir):
+def get_user_defined_globals(globals_str):
+    """Parse user-defined globals into an immutable dict using globals_str
+
+    :param globals_str Either a file, in which case, attempt to open the file and parse the contents as JSON,
+        or a JSON string representing a JSON object. The parsed JSON must represent a collection of key-value pairs,
+        i.e. a python dict.
+    :return dict containing user-defined global variables
+    """
+    if globals_str is None:
+        return pysistence.make_dict()
+
+    from_file = False
+    if os.path.isfile(globals_str):
+        # The string appears to be a file, so try loading JSON from file
+        # This may raise an IOError if the file can't be read or a ValueError if the contents of the file
+        # cannot be parsed.
+        print "globals parameter appears to be a file. Attempting to load user defined globals from file %s..." % globals_str
```

Comment:
I thought this was just debugging code though. Do we even need to be
printing this at all?

On Wed, Apr 27, 2016, 5:19 PM Geoff <notifications@github.com> wrote:

> In ducktape/command_line/main.py
> <https://github.com/confluentinc/ducktape/pull/135#discussion_r61358245>:
>
> > +    """Parse user-defined globals into an immutable dict using globals_str
> > +
> > +    :param globals_str Either a file, in which case, attempt to open the file and parse the contents as JSON,
> > +        or a JSON string representing a JSON object. The parsed JSON must represent a collection of key-value pairs,
> > +        i.e. a python dict.
> > +    :return dict containing user-defined global variables
> > +    """
> > +    if globals_str is None:
> > +        return pysistence.make_dict()
> > +
> > +    from_file = False
> > +    if os.path.isfile(globals_str):
> > +        # The string appears to be a file, so try loading JSON from file
> > +        # This may raise an IOError if the file can't be read or a ValueError if the contents of the file
> > +        # cannot be parsed.
> > +        print "globals parameter appears to be a file. Attempting to load user defined globals from file %s..." % globals_str
>
> @ewencp <https://github.com/ewencp>
> There are already a couple of other print calls restricted to the main
> module which are only used when it's about to exit since the
> session_context object is not available throughout.
>
> I figure either we add another logger instance here, or just print.
> Thoughts?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/confluentinc/ducktape/pull/135/files/1b488c563bb744f5d51dbdb33b2381dcf9f51776#r61358245>
>